pipit/readers/nsight_reader.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2022 Parallel Software and Systems Group, University of Maryland.
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+
+import pandas as pd
+import pipit.trace
+
+
+class NsightReader:
+    """Reader for NSight trace files"""
+
+    def __init__(self, file_name) -> None:
+        self.file_name = file_name
+        self.df = None
+
+    """
+    This read function directly takes in a csv of the trace report and
+    utilizes pandas to convert it from a csv into a dataframe.
+    """
+
+    def read(self):
+        # Read in csv
+        self.df = pd.read_csv(self.file_name)
+
+        # Copy data into new dataframe
+        df = self.df
+
+        pid, tid = set(df["PID"]), set(df["TID"])
+
+        df = df.astype(
+            {
+                "PID": "category",
+                "TID": "category",
+            }
+        )
+
+        # check if multi-process, single-threaded trace, if so remove tid column
```

Comment:
check if the PID and TID column has the same values, if so drop the TID column (because num_threads_per_process = 1)
then check if all values in PID column are the same